### PR TITLE
Using Topic CRD API instead of ConfigMap for Strimzi Topic Operator

### PR DESCRIPTION
**Pull request by matzew**
Tested w/ Strimzi 0.7.0 and worked:

* tested manual creation of two topics (`kubectl create -f ./my-cr.yml`), one had the `source` label, one not.

_RESULT_ only the topic w/o our label showed up :heavy_check_mark: 

* created a flow from the archetype, and saw that `kafkatopic` CRD API was used, e.g. in the logs..

Saw (`oc get kafkatopic`) that one was created - here is the yaml:
```
oc get kafkatopic sf1-401c1943-a74d-8adf-a637-c9155d34e881-output-data -o yaml 
```

Output:
```
apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaTopic
metadata:
  annotations: {}
  clusterName: ""
  creationTimestamp: 2018-09-20T08:23:24Z
  finalizers: []
  generation: 1
  labels:
    app: sf1
    streamzi.io/source: autocreated
    strimzi.io/cluster: my-cluster
  name: sf1-401c1943-a74d-8adf-a637-c9155d34e881-output-data
  namespace: myproject
  ownerReferences: []
  resourceVersion: "149006"
  selfLink: /apis/kafka.strimzi.io/v1alpha1/namespaces/myproject/kafkatopics/sf1-401c1943-a74d-8adf-a637-c9155d34e881-output-data
  uid: 70f252d4-bcae-11e8-a505-c85b765a6fb1
spec:
  config: {}
  partitions: 20
  replicas: 1
```

### CODE REVIEW

**Comment by matzew** [comment]
one thing, the Strimzi API itself adds it's bits now to the `label` section - so did I

File: model/src/main/java/io/streamzi/openshift/dataflow/deployment/TargetState.java [hunk 1]
```diff
@@ -232,30 +254,28 @@ private ConfigMap buildEnvVarConfigMaps(ProcessorNode node) {
...
 
-    private ConfigMap buildTopicConfigMaps(ProcessorLink link) {
-
-        String topicName = flow.getName() + "-" + link.getSource().getParent().getUuid() + "-" + link.getSource().getName();
-
-        final Map<String, String> data = new HashMap<>();
-        data.put("name", topicName);
-        data.put("partitions", "20");
-        data.put("replicas", "1");
+    private CustomResource buildTopicCrds(ProcessorLink link) {
```
Comment (sjwoodman) [inline model/src/main/java/io/streamzi/openshift/dataflow/deployment/TargetState.java]: Very picky but shouldn't this be `buildTopicCrs` ?

Comment (matzew) [reply model/src/main/java/io/streamzi/openshift/dataflow/deployment/TargetState.java]: make sense !

or even `buildTopicCr` ?

Comment (sjwoodman) [reply model/src/main/java/io/streamzi/openshift/dataflow/deployment/TargetState.java]: +1

**Pull request merged**